Furball.Vixie.Backends.Direct3D11/LineRendererD3D11.cs

```diff
@@ -0,0 +1,260 @@
+using System.Runtime.InteropServices;
+using System.Text;
+using Furball.Vixie.Backends.Shared;
+using Furball.Vixie.Backends.Shared.Renderers;
+using Furball.Vixie.Helpers.Helpers;
+using Vortice.D3DCompiler;
+using Vortice.Direct3D;
+using Vortice.Direct3D11;
+using Vortice.DXGI;
+
+namespace Furball.Vixie.Backends.Direct3D11 {
+    public class LineRendererD3D11 : ILineRenderer {
+        public bool IsBegun { get; set; }
+
+        private Direct3D11Backend   _backend;
+        private ID3D11Device        _device;
+        private ID3D11DeviceContext _deviceContext;
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct VertexData {
+            public Vector2 Position;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct InstanceData {
+            public Vector2 InstancePosition;
+            public Vector2 InstanceSize;
+            public Color   InstanceColor;
+            public float   InstanceRotation;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct ConstantBufferData {
+            public Matrix4x4 ProjectionMatrix;
+        }
+
+        private const int VERTEX_BUFFER_SLOT = 0;
+        private const int INSTANCE_BUFFER_SLOT = 1;
+
+        private const int INSTANCE_AMOUNT = 8192;
+
+        private ID3D11InputLayout  _inputLayout;
+        private ID3D11VertexShader _vertexShader;
+        private ID3D11PixelShader  _pixelShader;
+        private ID3D11Buffer       _vertexBuffer;
+        private ID3D11Buffer       _indexBuffer;
+        private ID3D11Buffer       _constantBuffer;
+        private ID3D11Buffer       _instanceBuffer;
+
+        private int            _instances;
+        private InstanceData[] _instanceData;
+
+        public unsafe LineRendererD3D11(Direct3D11Backend backend) {
+            this._backend       = backend;
+            this._device        = backend.GetDevice();
+            this._deviceContext = backend.GetDeviceContext();
+
+            string shaderSource = ResourceHelpers.GetStringResource("Shaders/LineRenderer/Shaders.hlsl");
+
+            Compiler.Compile(shaderSource, Array.Empty<ShaderMacro>(), null, "VS_Main", "VertexShader.hlsl", "vs_5_0", ShaderFlags.EnableStrictness, EffectFlags.None, out Blob vertexShaderBlob, out Blob vertexShaderErrorBlob);
+            Compiler.Compile(shaderSource, Array.Empty<ShaderMacro>(), null, "PS_Main", "PixelShader.hlsl", "ps_5_0", ShaderFlags.EnableStrictness, EffectFlags.None, out Blob pixelShaderBlob, out Blob pixelShaderErrorBlob);
+
+            //if (vertexShaderErrorBlob != null)
+            //    throw new Exception("LineRendererD3D11 Vertex Shader failed to compile! Error Log:\n" + Encoding.UTF8.GetString(vertexShaderErrorBlob.AsBytes()));
+//
+            //if (pixelShaderErrorBlob != null)
+            //    throw new Exception("LineRendererD3D11 Pixel Shader failed to compile! Error Log:\n" + Encoding.UTF8.GetString(pixelShaderErrorBlob.AsBytes()));
```

Comment:
Why are these commented out?

Reply:
cuz it annoyed me cuz even warnings were getting picked up, ill replace it with the error checking that the other ones do, so that only when it truly fails to compile this will trigger